task_mapper/heft_mulhome/original/read_input_heft.py

```diff
@@ -95,6 +95,7 @@ def get_exec_profile_data(exec_home_ip, MONGO_SVC_PORT, num_nodes):
             num_profilers = len(collection)
             logging.debug('Current number of updated execution profilers:')
             logging.debug(num_profilers)
+            time.sleep(3)
```

Comment:
Depending on lag, the needed sleep time maybe longer. Is there a way to make this event based? Or is a 3 second sleep guaranteed to work?

Reply:
I think it doesn't matter how many seconds to wait, it will keep retrying anyway, I just don't want it to print so many lines under this "busy while", which will ruin our reading of the HEFT log

Reply:
Does this actually keep looping? I see an exception  clause which will sleep for 60 seconds